src/wemmick/grpc_server.py

```diff
@@ -0,0 +1,58 @@
+from concurrent import futures
+import grpc
+from grpc_reflection.v1alpha import reflection
+import wemmick_pb2
+import wemmick_pb2_grpc
+import api
+
+
+class Wemmick(wemmick_pb2_grpc.WemmickServicer):
+
+    def HelloWorld(self, request, context):
+        return wemmick_pb2.Response(message='Hello World!')
+
+    def ListDataSources(self, request, context):
+        data_sources = api.list_datasources()
+        result = []
+        for data_source_item in data_sources:
+            credentials = wemmick_pb2.DataSource.Credentials(
+                connection_string=data_source_item['credentials']['connection_string']
+            )
+            data_source = wemmick_pb2.DataSource(
+                name=data_source_item['name'],
+                class_name=data_source_item['class_name'],
+                module_name=data_source_item['module_name'],
+                credentials=credentials
```

Comment:
We probably don't want to publish credentials.

Reply:
Yup I remember the discomfort as I wrote that line. I added a todo in the description as wanted to decide if we drop this altogether or use some sort or regex to mask the secrets but not the host and username.

The credentials definition is also incomplete. It currently only expects `connection_url`.

Reply:
I think wemmick can check env and creds file:
 
ENV=prod|dev|ect
CREDENTIALS_FILE_PATH=creds.yaml

As for consumers of wemmick, they can vol mount or build a new image.  I want to use sops for secret change management via git. kan-428

BUT this is an open source project, and we dont put any db config/cred in it.

Reply:
This api returns the datasource's connection_url (nested in credentials). I was hoping to parse the connection_url and mask the secret while returning the db name and table name.  I removed the credentials field for now. Will have to leave that for later.